### DISCUSSION TITLE
"Open Full" button to drawer

### DISCUSSION THREAD

**Discussion by LevinGermann** [directus/directus#8635] [Feature Requests]
Hello guys

I raised this issue in Discord and was asked to leave a Feature Request here.
I would like to have a button on the drawer that opens the current "mini-view" in a full window, like if you would open it directly from a collection.

The reason for this is that we have some quite nested data, and we got some feedback from our content writers that the current functionality with opening over opening child-data becomes confusing over time to know in what context the nested data is currently written.
I think if there would be a possibility to open the current open drawer-view as the new parent, it would be more clear to edit nested data again.

Let me know what you think.

Best regards

**Comment by danieljwestman** [comment]
I've had the same feedback/complaints from users. When having multiple nested items/relations it gets pretty confusing. Maybe some kind of "breadcrumps" in the top of the drawer would be helpful.

**Reply by benhaynes** [reply]
This mockup provides two new styles, a vertical label of layered drawers, and a listing of the parent and drawers currently open. We could implement one or both of these design options. @rijkvanzanten thoughts?

![Item Drawers](https://user-images.githubusercontent.com/522079/136577244-21725709-ddc9-41db-8d3e-2e8007535de3.png)

**Comment by benhaynes** [comment]
One thing to keep in mind is that if you open a detail page in full (not in a drawer) that you will be migrating away from the parent item, and will have to save/discard your work first. This is just how the system works (and most other systems).

I still think we can do some things to improve this experience, though. Such as the breadcrumb that @danieljwestman mentioned (I'll mock up a design for that now). What else is "missing" from the drawer experience?

**Reply by rijkvanzanten** [reply]
> What else is "missing" from the drawer experience?

Another thing that could help is a better nested visual style, like the POC I did in https://github.com/directus/directus/pull/6440

**Reply by danieljwestman** [reply]
Love the nested drawers PoC ❤️ 

That and maybe some indication/breadcrumbs in the header.

**Reply by rijkvanzanten** [reply]
> One thing to keep in mind is that if you open a detail page in full (not in a drawer) that you will be migrating away from the parent item, and will have to save/discard your work first. This is just how the system works (and most other systems).

Btw, one more thing to add on to that is that you may or may not have access to the module the item lives in 🤔 If you open a drawer for a related "user", the "open as full page" interaction would take you to the users module, which may or may not be accessible to your user as per that other discussion around role-based module de-registration

**Reply by LevinGermann** [reply]
> One thing to keep in mind is that if you open a detail page in full (not in a drawer) that you will be migrating away from the parent item, and will have to save/discard your work first. This is just how the system works (and most other systems).
> 
> I still think we can do some things to improve this experience, though. Such as the breadcrumb that @danieljwestman mentioned (I'll mock up a design for that now). What else is "missing" from the drawer experience?

One thing that came to my mind why this might be needed as well: currently if users work with nested collections, if a nested object is opened the collaboration feature of Directus (e.g. comments) completely drops away because these are not accessible in the "open-in-current-view"-mode...

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
